src/data.c

```diff
@@ -1001,8 +1007,8 @@ static void poll_ctrl(void) {
 
         pr_response_flush(&resp_err_list);
 
-        destroy_pool(cmd->pool);
         pr_response_set_pool(resp_pool);
+        destroy_pool(cmd->pool);
```

Comment:
Here we make sure to set the Response API pool back _before_ destroying the pool for the interrupting command.